x-pack/platform/plugins/shared/alerting/server/rules_client/lib/extract_references.ts

```diff
@@ -45,10 +50,11 @@ export async function extractReferences<
     name: `${extractedSavedObjectParamReferenceNamePrefix}${reference.name}`,
   }));
 
-  const references = [...actionReferences, ...paramReferences];
+  const references = [...actionReferences, ...paramReferences, ...artifactReferences];
```

Comment:
Is there any deduplication of references that should be happening here? Pretty sure actionReferences and paramsReferences wouldn't include the types of saved object references we have in artifacts, but wondering if this will be necessary at some point.

Reply:
Good point... can you please review 7058cc8441dc22125bfd4a5389824fd2a029f654, I have added a call to `uniqBy` and added a test to ensure this gives the desired effect. Any references with equivalent ID field will be dropped after the first one.